src/vehicle.cpp

```diff
@@ -5182,11 +5182,14 @@ bool vehicle::fire_turret (int p, bool /* burst */ )
         if( liquid_fuel < 1 || charges < 1 ) {
             return false;
         }
-        itype *am_type = item::find_type( amt );
+        itype *am_type = item::find_type( default_ammo( amt ) );
+        itype *fake;
+        bool fake_ammo = false;
         if( !am_type->ammo ) {
-            // Let non-ammo-typed ammo be used too
-            // 0 in all stats, depends on gun to provide everything
-            am_type = item::find_type( "fake_ammo" );
+            // Alter ammo type temporarily
+            fake = item::find_type( "fake_ammo" );
+            std::swap( am_type->ammo, fake->ammo );
```

Comment:
Is it possible to have a "dual typed" item defined from json level?
From what I've seen, despite the new system handling all sorts of combinations of item classes, the json loaders only load a particular type and make it a separate itype.

Would it be possible (without reinventing a lot) to use two loaders for a single item without creating two itypes? Or at least merge them somehow without getting hacky.

If the water cannon is supposed to be a water cannon, it cannot really shoot a different type - I can't even add a type substitution for non-ammo items, or RECOVERY_x would drop the substituted type. The other solution I see would be getting into fire_gun and substituting things there, which would be lengthy and probably a fair bit of an overdesign for such a tiny thing.